lib/active_model/serializer/adapter/attributes.rb

```diff
@@ -2,42 +2,73 @@ module ActiveModel
   class Serializer
     module Adapter
       class Attributes < Base
+        attr_accessor :_nested_associations
+
+        def initialize(serializer, options = {})
+          options ||= {}
+
+          self._nested_associations = ActiveModel::Serializer::Utils
+            .include_args_to_hash(options[:include])
+
+          super(serializer, options)
+        end
+
         def serializable_hash(options = nil)
           options ||= {}
+
+          unless _nested_associations.present?
+            # not set in the initializer when rendered from a controller
+            self._nested_associations = ActiveModel::Serializer::Utils
+              .include_args_to_hash(options[:include])
+          end
+
           if serializer.respond_to?(:each)
-            result = serializer.map { |s| Attributes.new(s).serializable_hash(options) }
+            serializable_hash_for_collection(options)
           else
-            hash = {}
+            serializable_hash_for_single_resource(options)
+          end
+        end
+
+        def serializable_hash_for_collection(options = nil)
+          serializer.map { |s| Attributes.new(s).serializable_hash(options) }
+        end
```

Comment:
Man, I'd so prefer to have the adapter refactor separate from the work in this PR...